spec/lib/pdf_filler/md502_pdf_spec.rb

```diff
@@ -0,0 +1,145 @@
+require 'rails_helper'
+
+RSpec.describe PdfFiller::Md502Pdf do
+  include PdfSpecHelper
+
+  let(:intake) { create(:state_file_md_intake) }
+  let(:submission) { create :efile_submission, tax_return: nil, data_source: intake }
+  let(:pdf) { described_class.new(submission) }
+
+  describe '#hash_for_pdf' do
+    let(:file_path) { described_class.new(submission.reload).output_file.path }
+    let(:pdf_fields) { filled_in_values(file_path) }
+
+    it 'uses field names that exist in the pdf' do
+      missing_fields = pdf.hash_for_pdf.keys.map { |k| k.to_s.gsub("'", "&apos;").to_s } - pdf_fields.keys
+      expect(missing_fields).to eq([])
+    end
+
+    # We usually expect "Yes" to be the "checked" option in PDFs, but for this field "No" means checked.
+    # This test is to ensure when fixtures change, change is made to the corresponding has_for_pdf value
+    # on this field from "No" to "Yes"
+    it "pdf contains 'No' option for mfs checkbox" do
+      expect(check_pdf_option(file_path, "Check Box - 3", "No")).to eq(true)
+    end
+
+    context "single" do
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Enter day and month of Fiscal Year beginning']).to eq '01-01'
+          expect(pdf_fields['Enter day and month of Fiscal Year Ending']).to eq "12-31"
+          expect(pdf_fields['Enter social security number']).to eq("123456789")
+          expect(pdf_fields["Enter spouse's social security number"]).to be_nil
```

Comment:
[DUST] this is so silly but could we just make everything double quotes for consistence?

Reply:
this bothers me too. should've cleaned it up. ty!